keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/Optimizer.java

```diff
@@ -84,4 +88,19 @@ static long totalNumberOfLatentDimensions(List<? extends Vertex<? extends Number
     static long numDimensions(Vertex<? extends Tensor> vertex) {
         return vertex.getValue().getLength();
     }
+
+    static ProgressBar createFitnessProgressBar(final Optimizer optimizerThatNeedsProgressBar) {
+        AtomicInteger evalCount = new AtomicInteger(0);
+        ProgressBar progressBar = new ProgressBar();
+        BiConsumer<double[], Double> progressBarFitnessCalculationHandler = (position, logProb) -> {
+            progressBar.progress(
+                String.format("Fitness Evaluation #%d LogProb: %.2f", evalCount.incrementAndGet(), logProb)
```

Comment:
This is cool! Consider adding in the starting LogProb to the output too so people can see it increasing / how much it has increased?